plonk-hashing/src/lib.rs

```diff
@@ -6,7 +6,12 @@
 
 //! PLONK Hashing Library
 
-#![cfg_attr(not(any(feature = "std", test)), no_std)]
+// #![cfg_attr(not(any(feature = "std", test)), no_std)]
 #![cfg_attr(doc_cfg, feature(doc_cfg))]
 #![forbid(rustdoc::broken_intra_doc_links)]
-#![forbid(missing_docs)]
+// #![forbid(missing_docs)]
+
+#[macro_use]
+pub extern crate ark_std;
```

Comment:
We should not depend on `ark_std` and use the `core` and `alloc` libraries instead.

Reply:
Okay, took care of this.  There were two spots ([here](https://github.com/ZK-Garage/plonk/blob/097353b1eec48eb3bb083c44342b8535a1e3d335/plonk-hashing/src/poseidon/mds.rs#L3) and [here ](https://github.com/ZK-Garage/plonk/blob/097353b1eec48eb3bb083c44342b8535a1e3d335/plonk-hashing/src/poseidon/preprocessing.rs#L4)) where Manta's implementation uses `ark_ff::vec` and I was unsure whether to remove these.  It compiles fine without these imports--can you confirm that it's safe to remove them?

Reply:
The `ark_ff::vec` should be removed in favor of `alloc::vec`. It's just a re-export.